### DISCUSSION TITLE
Chart not showing data correctly

### DISCUSSION THREAD

**Discussion by pldmayer** [Budibase/budibase#1498] [Help]
Hi, I am trying to implement a chart into the screen, I have a external Mysql DB with a simple table
![Screenshot 2021-05-17 194226](https://user-images.githubusercontent.com/33716628/118533849-da202480-b748-11eb-9474-4b616de72c13.png)
Then I write a query ordered by ID
![Screenshot 2021-05-17 194145](https://user-images.githubusercontent.com/33716628/118533913-ee642180-b748-11eb-81c1-656fee8b13df.png)
but when I link the query to the chart the labels get mixed up
![Screenshot 2021-05-17 194250](https://user-images.githubusercontent.com/33716628/118533977-03d94b80-b749-11eb-9aa5-9ce6576d13ad.png)

Any Idea how I can resolve this issue??

**Comment by mjashanks** [comment]
Hmm, it seems like it is ordering alphabetically. To work around this, you could use SQL to convert the month values to `2021-04`, `2021-05` (alphabetically ordered months). Not ideal, but it will get you moving forward for now.

**Comment by aptkingston** [comment]
I replied this morning in the issue you created here: https://github.com/Budibase/budibase/issues/1499, but I'll copy my response from there:

Hi @pldmayer. Currently the charts sort data by their X axis (label column) which is usually desired for time series data, but not in all cases as you've highlighted. Our release coming out next week has huge improvements in it for searching and sorting data, which will let you sort your data however you want and have that properly reflected in charts.
